src/mongoose_hooks.erl

```diff
@@ -1063,125 +1063,138 @@ room_new_affiliations(Acc, Room, NewAffs, Version) ->
+               default_mode => DefaultMode, always_jids => AlwaysJIDs, never_jids => NeverJIDs},
+    Args = [HostType, ArchiveID, Owner,
+            DefaultMode, AlwaysJIDs, NeverJIDs],
     ParamsWithLegacyArgs = ejabberd_hooks:add_args(Params, Args),
-    run_hook_for_host_type(mam_set_prefs, HookServer, {error, not_implemented},
+    run_hook_for_host_type(mam_set_prefs, HostType, {error, not_implemented},
                            ParamsWithLegacyArgs).
 
 %%% @doc The `mam_get_prefs' hook is called to read
 %%% the archive settings for a given user.
--spec mam_get_prefs(HookServer, DefaultMode, ArchiveID, OwnerJID) -> Result when
-      HookServer :: jid:lserver(),
+-spec mam_get_prefs(HostType, DefaultMode, ArchiveID, Owner) -> Result when
+      HostType :: jid:lserver(),
       DefaultMode :: mod_mam:archive_behaviour(),
       ArchiveID :: undefined | mod_mam:archive_id(),
-      OwnerJID :: jid:jid(),
+      Owner :: jid:jid(),
       Result :: mod_mam:preference() | {error, Reason :: term()}.
-mam_get_prefs(HookServer, DefaultMode, ArchiveID, OwnerJID) ->
-    Params = #{archive_id => ArchiveID, owner_jid => OwnerJID},
-    Args = [HookServer, ArchiveID, OwnerJID],
+mam_get_prefs(HostType, DefaultMode, ArchiveID, Owner) ->
+    Params = #{archive_id => ArchiveID, owner => Owner},
+    Args = [HostType, ArchiveID, Owner],
     ParamsWithLegacyArgs = ejabberd_hooks:add_args(Params, Args),
     InitialAccValue = {DefaultMode, [], []}, %% mod_mam:preference() type
-    run_hook_for_host_type(mam_get_prefs, HookServer, InitialAccValue,
+    run_hook_for_host_type(mam_get_prefs, HostType, InitialAccValue,
                            ParamsWithLegacyArgs).
 
 %%% @doc The `mam_remove_archive' hook is called in order to
 %%% remove the entire archive for a particular user.
--spec mam_remove_archive(HookServer, ArchiveID, OwnerJID) -> any() when
-      HookServer :: jid:lserver(),
+-spec mam_remove_archive(HostType, ArchiveID, Owner) -> any() when
+      HostType :: jid:lserver(),
       ArchiveID :: undefined | mod_mam:archive_id(),
-      OwnerJID :: jid:jid().
-mam_remove_archive(HookServer, ArchiveID, OwnerJID) ->
-    Params = #{archive_id => ArchiveID, owner_jid => OwnerJID},
-    Args = [HookServer, ArchiveID, OwnerJID],
+      Owner :: jid:jid().
+mam_remove_archive(HostType, ArchiveID, Owner) ->
+    Params = #{archive_id => ArchiveID, owner => Owner},
+    Args = [HostType, ArchiveID, Owner],
     ParamsWithLegacyArgs = ejabberd_hooks:add_args(Params, Args),
-    run_hook_for_host_type(mam_remove_archive, HookServer, ok,
+    run_hook_for_host_type(mam_remove_archive, HostType, ok,
                            ParamsWithLegacyArgs).
 
 %%% @doc The `mam_lookup_messages' hook is to retrieve
 %%% archived messages for given search parameters.
--spec mam_lookup_messages(HookServer, Params) -> Result when
-      HookServer :: jid:lserver(),
+-spec mam_lookup_messages(HostType, Params) -> Result when
+      HostType :: jid:lserver(),
```

Comment:
If this is a `HostType`, the type name could also change to `mongooseim:host_type()`. Same applies to other hooks below.